src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -341,17 +322,16 @@ public SearchOptions Create(string compartmentType, string compartmentId, string
                     catch (SearchParameterNotSupportedException)
                     {
                         sortingsValid = false;
-                        _contextAccessor.RequestContext?.BundleIssues.Add(new OperationOutcomeIssue(
-                            OperationOutcomeConstants.IssueSeverity.Warning,
-                            OperationOutcomeConstants.IssueType.NotSupported,
-                            string.Format(CultureInfo.InvariantCulture, Core.Resources.SearchParameterNotSupported, sorting.Item1, string.Join(", ", resourceTypesString))));
+                        searchSortErrors.Add(string.Format(CultureInfo.InvariantCulture, Core.Resources.SearchParameterNotSupported, sorting.Item1, string.Join(", ", resourceTypesString)));
                     }
                 }
 
+                // Check if sort parameters are valid and can be used for sorting.
                 if (sortingsValid)
                 {
                     if (!_sortingValidator.ValidateSorting(sortings, out IReadOnlyList<string> errorMessages))
                     {
+                        // Sanity check, ValidateSorting must output errors if it returns false.
```

Comment:
+1 thanks for adding comments to this method - I found it makes it easier to digest